webcurator-ui/src/views/target/target-tabs/TargetTabPanelGeneral/modals/TargetAddHarvestAuthModal.vue

```diff
@@ -1,88 +1,152 @@
 <script setup lang="ts">
 import { inject, onMounted, ref } from 'vue';
+import { formatDate } from '@/utils/helper';
 import { type UseFetchApis, useFetch } from '@/utils/rest.api';
 
 const dialogRef: any = inject('dialogRef');
 
 const rest: UseFetchApis = useFetch();
 
-const harvestAuths = ref([]);
-const filteredHarvestAuths = ref([]);
+interface HarvestAuth {
+  id: number;
+  name: string;
+  authorisingAgents: any[];
+  permissions: any[];
+}
+
+const returnedHarvestAuths = ref<HarvestAuth[]>([]);
 const loading = ref(false);
 
+const harvestAuths = ref<{ id: number, name: string, agent: '', permissionId: number, startDate: '', endDate: '', urlPatterns: [] }[]>([]);
+const filteredHarvestAuths = ref<{ id: number, name: string, agent: '', permissionId: number, startDate: '', endDate: '',  urlPatterns: []  }[]>([]);
+
 const searchTerm = ref('');
 
 const seed = ref();
 
-const fetch = () => {
-    const searchParams = {
-        offset: 0,
-        limit: 1024
+const prepareData = (data: HarvestAuth[]) => {
+  data.forEach((harvestAuth: HarvestAuth) => {
+    if (harvestAuth.authorisingAgents.length > 0) {
+      harvestAuth.authorisingAgents.forEach((authorisingAgent: any) => {
+        if (authorisingAgent.permissions.length > 0) {
+          harvestAuths.value.push({
+            id: harvestAuth.id,
+            name: harvestAuth.name,
+            agent: authorisingAgent.name,
+            permissionId: authorisingAgent.permissions[0].id,
+            startDate: authorisingAgent.permissions[0].startDate,
+            endDate: authorisingAgent.permissions[0].endDate,
+            urlPatterns: authorisingAgent.permissions[0].urlPatterns
+          });
+        }
+      })
     }
+  });
+  filteredHarvestAuths.value = harvestAuths.value;
+};
 
-    loading.value = true;
-
-    rest
-        .post('harvest-authorisations', searchParams, { header: 'X-HTTP-Method-Override', value: 'GET' })
-        .then((data: any) => {
-            harvestAuths.value = data['harvestAuthorisations']
-            filteredHarvestAuths.value = harvestAuths.value
-            loading.value = false
-        })
-        .catch((err: any) => {
-            console.log(err.message)
-            loading.value = false
-        })
-}
+const fetch = () => {
+  const searchParams = {
+    offset: 0,
+    limit: 1024,
+  };
+
+  loading.value = true;
+
+  rest
+    .post('harvest-authorisations', searchParams, { header: 'X-HTTP-Method-Override', value: 'GET' })
+    .then((data: any) => {
+      returnedHarvestAuths.value = data['harvestAuthorisations'];
+      prepareData(returnedHarvestAuths.value);
+      loading.value = false;
+    })
+    .catch((err: any) => {
+      console.log(err.message);
+      loading.value = false;
+    });
+};
 
 const search = () => {
```

Comment:
If there is no filter field defined implicitly, the "Authorising Agent", "Start Date", "End Date" can be taken as filtered conditions as well.

Reply:
Added a tooltip to the search input to explain that is searches names and url patterns